File: Data Structures & Algorithms/string-encode-and-decode/submission-8.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        s=""
        for i in strs:
            s+=str(len(i))+"#"+i
        return s
            

    def decode(self, s: str) -> List[str]:
        ans = []
        l = 0
        while l<len(s):
            r=l
            while s[r]!="#":
                r+=1
            i = int(s[l:r])
            t = s[r+1: r+i+1]
            ans.append(t)
            l=r+i+1

        return ans

File: Data Structures & Algorithms/anagram-groups/submission-8.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const m={}
        const ans=[]
        for(let s of strs){
            const c=new Array(26).fill(0)
            for(let i=0;i<s.length;i++){
                c[s.charCodeAt(i)-"a".charCodeAt(0)]++
            }
            if(!m[c]) m[c]=[];
            m[c].push(s)
        }
        for(let i in m){
            ans.push(m[i])
        }
        return ans
    }
}
